pg_search/src/postgres/vacuum.rs

```diff
@@ -17,10 +17,15 @@
 
 use pgrx::*;
 
+use crate::postgres::merge::do_merge;
+use crate::postgres::rel::PgSearchRelation;
+
 #[pg_guard]
 pub unsafe extern "C-unwind" fn amvacuumcleanup(
-    _info: *mut pg_sys::IndexVacuumInfo,
+    info: *mut pg_sys::IndexVacuumInfo,
     stats: *mut pg_sys::IndexBulkDeleteResult,
 ) -> *mut pg_sys::IndexBulkDeleteResult {
+    let index = PgSearchRelation::open((*(*info).index).rd_id);
+    do_merge(&index).expect("should be able to merge");
```

Comment:
calling `do_merge()` here means it could merge in both the foreground and the background.

I do not believe that VACUUM should be doing any in-process (foreground) merging.  a) we need to get and out of VACUUM as quickly as possible, and b) I am not sure it's even correct.

Perhaps `do_merge()` needs an enum flag for the type of merging the caller wants it to do.

```rust
enum MergeStyle {
   Foreground,
   Background,
   Bothground, // I dunno, pick a name
}